### DISCUSSION TITLE
transmitting a database as a struct

### DISCUSSION THREAD

**Discussion by sto1cLem0n5** [jgromes/RadioLib#225] [Q&A] [answered]
Hi, I want to send a database of information to neighbouring nodes in a network so that eventually, all the nodes have a complete database with all the data from all the nodes. I believe that the function `transmit(uint8_t* data, size_t len, uint8_t addr)` can only take a `uint8_t` variable. My database struct similar to this:
```
databaseEntry sendData[20]; // this is the variable I want to input to transmit()

typedef struct
{
  uint16_t nodeID;
  uint8_t event;
  uint32_t timestamp;
} databaseEntry;
```

Is my goal possible to achieve? If not, what other ways could I transmit larger sizes of data in LoRa?

**Comment by jgromes** [comment, marked as answer]
> can only take a uint8_t variable

No, it takes an uint8_t **pointer** - so it can send anything if you cast it to that pointer. Which makes sense - after all, your struct is just a bunch of bytes, which is what the radio is sending.

It's perfectly fine to do the following (if some it is unclear, I suggest reading up on pointers):

```c++
typedef struct
{
  uint16_t nodeID;
  uint8_t event;
  uint32_t timestamp;
} databaseEntry;

databaseEntry entry = {
  .nodeID = 1,
  .event = 2,
  .timestamp = 3,
}

uint8_t* entryPtr = (uint8_t*)&entry;
transmit(entryPtr, sizeof(databaseEntry));
```
